### PR TITLE
analysis: do Swiss officers have more mutual connections to abroad officers than other Swiss officers?

### LINKED ISSUE #12: analysis: do Swiss officers have more mutual connections to abroad officers than other Swiss officers?
Entities können auch mehr als einen Officer besitzen.
Teilen sich Schweizer Officers mehr Firmen (Entities) mit anderen Schweizer Officers oder mehr mit ausländischen?

Gemischt Dyadischer-Monadischer Hypothesentest?

### PR DESCRIPTION
**Pull request by buesst1**
- Correlation between country and interaction between officers
- Ratio between number of connections to swiss officers and degree respective Swiss officers

Closes #12

### CODE REVIEW

File: 05-swiss-officers-more-introverted.ipynb [hunk 1]
```diff
@@ -0,0 +1,684 @@
+{
```
Comment (florinbarbisch) [inline 05-swiss-officers-more-introverted.ipynb:1]: <p>QAP in dyadic_hypothesis_test is implemented wrong. See comment there. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gabrieltorresgamez/fhnw-san-project/pull/17/discussion/#comment-1608289460'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

File: utils.py [hunk 2]
```diff
@@ -480,3 +480,175 @@ def permute_graph_QAP(G: nx.Graph, self_loops=False) -> nx.Graph:
...
+        Perform a Quadratic Assignment Procedure (QAP) by randomly permuting the rows and columns of the adjacency matrix.
+
+        Parameters:
+        adj_matrix (torch.Tensor): The adjacency matrix to permute.
+
+        Returns:
+        torch.Tensor: The permuted adjacency matrix.
+        """
+        # Define random row and column permutations for G2_adj
+        row_permutation = torch.randperm(adj_matrix.shape[0], device=device)
```
Comment (florinbarbisch) [inline utils.py]: The row and column permuations NEED to be the same.

image

**Review by florinbarbisch — CHANGES_REQUESTED** [review]
Fix QAP

File: utils.py (hunk 2 above)
Comment (buesst1) [reply utils.py]: Right...

**Review by florinbarbisch — APPROVED** [review]

**Pull request merged**